staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go

```diff
@@ -294,20 +299,28 @@ func (o *DeleteOptions) RunDelete(f cmdutil.Factory) error {
 }
 
 func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
-	found := 0
 	if o.IgnoreNotFound {
 		r = r.IgnoreErrors(errors.IsNotFound)
 	}
+
+	deletedInfos, err := r.Infos()
+	if len(deletedInfos) == 0 {
+		fmt.Fprintf(o.Out, "No resources found\n")
+		return err
+	}
+	if o.Interactive && !o.confirmation(deletedInfos) {
+		fmt.Fprintf(o.Out, "deletion is cancelled\n")
+		return err
```

Comment:
What kind of error might we be returning here? I'm wondering what this might look like to the end user (e.g. if we print "deletion is cancelled" _followed_ by an error message that sounds like we did something else, that might be alarming)

Reply:
This is the preserve the same functionality provided by `Visit` function. Because it behaves exactly in this way.